File: node/src/entity/Item.ts
import {Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn} from "typeorm";
import {getManager} from "typeorm";
import { User } from "./User";

@Entity('items')
export class Item {
    @PrimaryGeneratedColumn()
    public id: number;

    @Column()
    public name: string;

    @Column()
    public price: number;

    @ManyToOne(type => User, user => user.items)
    @JoinColumn({ name: "user_id" })
    user: User | undefined;

    @CreateDateColumn({name: 'created_at', type: "timestamp"})
    public createdAt: Date;

    @UpdateDateColumn({name: 'updated_at',type: "timestamp"})
    public updatedAt: Date;

    constructor(name: string, price: number, user: User | undefined) {
        this.name = name
        this.price = price
        this.user = user
    }
}

File: kotlin/src/main/kotlin/io/github/devstarsj/kotlinbackend/Item.kt
package io.github.devstarsj.kotlinbackend

import com.fasterxml.jackson.annotation.JsonIgnore
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "items")
data class Item(
        @Id
        @GeneratedValue
        var id: Long? = null,
        var name: String,
        var price: Int,
        @Column(name="user_id")
        var userId: Long? = null,

        @ManyToOne(optional = true)
        @JoinColumn(name = "user_id", insertable = false, updatable = false)
        @JsonIgnore
        val user: User? = null,

        @CreationTimestamp
        @Column(name = "created_at")
        var createdAt: LocalDateTime = LocalDateTime.now(),
        @UpdateTimestamp
        @Column(name = "updated_at")
        var updatedAt: LocalDateTime = LocalDateTime.now()
)